crates/amaru-kernel/src/lib.rs

```diff
@@ -268,6 +271,35 @@ impl From<&[u8]> for RawBlock {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct TransactionInputAdapter(TransactionInput);
+
+impl Deref for TransactionInputAdapter {
+    type Target = TransactionInput;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::fmt::Display for TransactionInputAdapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}#{}", self.0.transaction_id, self.0.index)
+    }
+}
+
+impl From<TransactionInput> for TransactionInputAdapter {
+    fn from(value: TransactionInput) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&TransactionInput> for TransactionInputAdapter {
+    fn from(value: &TransactionInput) -> Self {
+        value.clone().into()
+    }
+}
```

Comment:
I'd rather not have that. That's a hidden clone, and we already have `From<TransactionInput> for TransactionInputAdapter`. So if you have a reference which needs transformation, it's better to let the caller clone.